src/components/KeyboardAwareScrollView/index.tsx

```diff
@@ -95,25 +109,167 @@ const KeyboardAwareScrollView = forwardRef<
+            "Goal:",
+            goal,
+            "Current:",
+            scrollOffsetY.value,
+            "StartingPosition:",
+            scrollY.value,
+          );
+
+          return;
+        }
+      }
+      if (goal === scrollOffsetY.value) {
+        log(
+          "Already at target position.",
+          "Goal:",
+          goal,
+          "Current:",
+          scrollOffsetY.value,
+          "StartingPosition:",
+          scrollY.value,
+        );
+
+        return;
+      }
+      cancelAnimation(scrollY);
+
+      scrollGoal.value = goal;
+
+      if (!animated) {
+        scrollGoal.value = goal;
+      }
+      isScrolling.value = true;
+      log(
+        "Goal:",
+        goal,
+        "Current:",
+        scrollOffsetY.value,
+        "StartingPosition:",
+        scrollY.value,
+        "Distance From Goal:",
+        Math.abs(scrollOffsetY.value - goal),
+        "Distance From Start:",
+        Math.abs(scrollOffsetY.value - scrollY.value),
+        "Is keyboard opening:",
+        keyboardWillAppear.value,
+        "is keyboard visible:",
+        isKeyboardVisible.value,
+        "scrollableHeight:",
+        scrollableHeight.value - scrollViewLayout.value.height,
+        "keyboardHeight:",
+        keyboardHeight.value,
+        "keyboardFrameHeight:",
+        currentKeyboardFrameHeight.value,
+      );
+
+      // Edge Case - Android Only
+      if (
+        Math.round(Math.abs(scrollOffsetY.value - goal)) ===
+          keyboardHeight.value &&
+        Math.round(scrollableHeight.value - scrollViewLayout.value.height) ===
+          Math.round(scrollOffsetY.value) &&
+        Platform.OS === "android"
+      ) {
+        log("Keyboard is at the bottom of the scroll view.");
+        // on Android, when at the bottom of the scroll view, it can not scroll
+        // to the goal position when keyboard is opening.
+        // This is a workaround to scroll to the goal position
+        scrollY.value = goal - keyboardHeight.value;
+      }
+      scrollY.value = spring
+        ? withSpring(goal, { damping: 100, stiffness: 100 }, (finished) => {
+            log(
+              "Goal:",
+              goal,
+              "Current:",
+              scrollOffsetY.value,
+              "EndingPosition:",
+              scrollY.value,
+              "Distance From Goal:",
+              Math.abs(scrollOffsetY.value - goal),
+              "Finished:",
+              finished,
+            );
+            isScrolling.value = false;
+          })
+        : withTiming(goal, undefined, (finished) => {
+            log(
+              "Goal:",
+              goal,
+              "Current:",
+              scrollOffsetY.value,
+              "EndingPosition:",
+              scrollY.value,
+              "Distance From Goal:",
+              Math.abs(scrollOffsetY.value - goal),
+              "Finished:",
+              finished,
+            );
+            isScrolling.value = false;
+          });
+    };
```

Comment:
Why do we need this? We should perform scroll based on a keyboard position? So we should scroll from `onMove` handlers.

Plain scroll can happen when text input grows, but in this case `scrollTo(..., true)` with default animation works also pretty well 🤔